### PR TITLE
feat: Add feature flag to allow/disallow participants with exceptions…

### PR DESCRIPTION
**Pull request by Joseph2910**
… to flow to cohort distribution

## Description

Added environment variable to allow participants with errors to be added to cohort distribution table as a feature for the testing team.
## Context
https://nhsd-jira.digital.nhs.uk/browse/DTOSS-6265

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [X] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [X] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [X] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs [hunk 1]
```diff
@@ -152,4 +152,10 @@ private bool ParticipantHasException(string nhsNumber, string screeningId)
         var exceptionFlag = Enum.TryParse(participant.ExceptionFlag, out Exists value) ? value : Exists.No;
         return exceptionFlag == Exists.Yes;
     }
+
+    private bool RetrieveEnivornmentalVariableAsBool(string environmentVariableName) {
```
Comment (MWClayson-NHS) [inline application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:156]: Please move the brackets to new lines. on this and the if statement below

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs [hunk 2]
```diff
@@ -152,4 +152,10 @@ private bool ParticipantHasException(string nhsNumber, string screeningId)
...
+        if (Environment.GetEnvironmentVariable(environmentVariableName) == "true") {
+            return true;
+        } else return false;
```
Comment (MWClayson-NHS) [inline application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:159]: There is no need for an else statement here you can just return false and the method will exit early.

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs (hunk 1 above)
Comment (MWClayson-NHS) [reply application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:156]: Can you also make this method static?

**Review by MWClayson-NHS — APPROVED** [review]

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs [hunk 3]
```diff
@@ -152,4 +152,13 @@ private bool ParticipantHasException(string nhsNumber, string screeningId)
         var exceptionFlag = Enum.TryParse(participant.ExceptionFlag, out Exists value) ? value : Exists.No;
         return exceptionFlag == Exists.Yes;
     }
+
+    private static bool RetrieveEnivornmentalVariableAsBool(string environmentVariableName)
```
Comment (andrewmorris43) [inline application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:156]: There's a typo in this method name.

**Review by andrewmorris43 — DISMISSED** [review]

**Comment by will-larkin-nhs** [comment]
Please add a proper description and a link to the ticket in the context

**Review by will-larkin-nhs — APPROVED** [review]

**Review by Warren-Pitterson — APPROVED** [review]

**Pull request merged**